### ISSUE TITLE
Building polygons Coalesce 

### ISSUE THREAD

**Issue by ghigio2000** [mapbox/tippecanoe#788]
Hi,

I am trying to merge adjacent buildings.

I used --coalesce with --hilbert reordering. 
Several buildings have been merged but most of them are still separated, even if they looks like to be adjacent.
It seems "coalesce" checks if two polygons are adjacent using the geoJSON geometry instead of the pixels coordinates zoom by zoom level. Isn't it?

Below screen shots show the buildings at the Z13 and Z16
![IMG_E6928 2](https://user-images.githubusercontent.com/4995563/63584199-bdb77e00-c59c-11e9-8ad5-08e1e188c799.JPG)
![IMG_6929 1](https://user-images.githubusercontent.com/4995563/63584219-c5772280-c59c-11e9-8638-aa46cfc5e594.PNG)

Is there a way to set a "tolerance" to let coalesce merge polygons are not 100% geometrically adjacent?

Thanks for your help
Giorgio

**Comment by ericfischer** [comment]
If you use `--reorder`, features with identical attributes will be reordered to be adjacent in the feature order and will be combined if you use `--coalesce`.

I should add an option to give more control over coalescing, but that is all that is available currently.

**Comment by ghigio2000** [comment]
I already used -x to remove all attributes.
So my understanding is correct? The coalesce works on the original geometry not on the pixels coordinates zoom by zoom, right?

**Comment by ericfischer** [comment]
No, the coalescing is done individually in each tile at each zoom level, not globally at the start of tiling.

**Comment by ghigio2000** [comment]
So it seems the reason why my sample does not coalesce because polygons are not ordered in a consecutive way.
Using -x there are no attributes to order by —reorder, so I used —hilbert 
Is there any other way to ensure a geographical reorder?

**Comment by ghigio2000** [comment]
Hi,
here attached is a test sample data.See the attached image red circle, I would expect, at low detailed zoom levels, to see these buildings collapse together into a single polygon or so.
You can find the JSON file and relative MBTILES generated with this command:
tippecanoe -o osmlanduse1-test.mbtiles -z14 -Z13 -s EPSG:3857 -x --hilbert --coalesce osmlanduse1-test.json
I also attached the shapefile the JSON come from (I have converted shape to json using ogr2ogr).
Thanks for your help,Giorgio

    Il lunedì 26 agosto 2019, 19:08:05 CEST, Eric Fischer <notifications@github.com> ha scritto:  
 
 
If you can provide a sample of your data, I can look into what is going wrong.

**Comment by ericfischer** [comment]
@ghigio2000 You don't seem to have included the link to the data in your reply, so I still can't debug this.

**Comment by ghigio2000** [comment]
The data was attached to the email
I will resend it as soon as back in office today

**Comment by ghigio2000** [comment]
Here attached is a zip file containing the JSON, relative MBTILES and the Shapefile created the JSON.
See also the attached Image with a red circle to concentrate the test on.

    Il giovedì 29 agosto 2019, 06:49:47 CEST, Ghigio <ghigio2000@yahoo.it> ha scritto:  
 
 The data was attached to the emailI will resend it as soon as back in office today 

Sent from my iPhone
On 29 Aug 2019, at 02:18, Eric Fischer <notifications@github.com> wrote:



@ghigio2000 You don't seem to have included the link to the data in your reply, so I still can't debug this.

**Comment by ericfischer** [comment]
Trying to attach files to Github tickets by email doesn't work. Please either include a link to the data or send it to me at fischer@mapbox.com.

**Comment by ericfischer** [comment]
When I do

```
tippecanoe -f -o osmlanduse1-test2.mbtiles -z14 -Z13 -s EPSG:3857 --reorder --hilbert --coalesce osmlanduse1-test.json
```

on your data, I am not seeing any obvious coalescing errors. I removed your `-x` and added `--reorder`.

(I think you really wanted `-X`, not `-x`, and that you were in fact excluding the (non-existent) attribute "--hilbert" with `-x --hilbert`, even though that wasn't what you meant.)

With neither `-x` nor `-X`, as above, the buildings with a `name` and the buildings with an `osm_id` are kept as separate features, while the other buildings that have no name or ID are merged into a single MultiPolygon in each tile.

If instead I do

```
tippecanoe -f -o osmlanduse1-test2.mbtiles -X -z14 -Z13 -s EPSG:3857 --reorder --hilbert --coalesce osmlanduse1-test.json
```

then all the attributes are removed and all the features in each tile are merged into a single MultiPolygon.

**Comment by ghigio2000** [comment]
Very good
I will test it today
Thanks
Giorgio
